### PR TITLE
Change ForAttributeWithMetadataName to use an explicit stack when recursing

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/64225

### CODE REVIEW

**Review by 333fred — APPROVED** [review]
Only a minor style comment.

File: src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs [hunk 1]
```diff
@@ -28,7 +28,8 @@ internal enum SourceGeneratorSyntaxTreeInfo
 
 public partial struct SyntaxValueProvider
 {
-    private static readonly ObjectPool<Stack<string>> s_stackPool = new ObjectPool<Stack<string>>(static () => new Stack<string>());
+    private static readonly ObjectPool<Stack<string>> s_stringStackPool = new(static () => new());
+    private static readonly ObjectPool<Stack<SyntaxNode>> s_nodeStackPool = new(static () => new());
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs:32]:
```suggestion
    private static readonly ObjectPool<Stack<string>> s_stringStackPool = new ObjectPool<Stack<string>>(static () => new Stack<string>());
    private static readonly ObjectPool<Stack<SyntaxNode>> s_nodeStackPool = new ObjectPool<Stack<SyntaxNode>>(static () => new Stack<SyntaxNode>());
```

[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs]: done.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @jaredpar @chsienki for another set of eyes on this incremental generator PR.  Thanks!

**Comment by CyrusNajmabadi** [comment]
https://github.com/orgs/dotnet/teams/roslyn-compiler @jaredpar @chsienki for another set of eyes on this incremental generator PR. Thanks!

**Review by jaredpar — APPROVED** [review]

**Comment by jaredpar** [comment]
All @CyrusNajmabadi changes need GitHub to default to clicking the "hide whitespace changes" button ;)

**Comment by CyrusNajmabadi** [comment]
Agreed :)

**Pull request merged**